source/agora/node/main.d

```diff
@@ -75,31 +76,47 @@ private int main (string[] args)
             writefln("Config file '%s' succesfully parsed.", cmdln.config_path);
             return 0;
         }
+        runTask(() => node = runNode(config));
```

Comment:
Note: This is the trick I was missing earlier. Because  `shutdown` needs to be called after the event loop returns (`listenHTTP` is not a blocking call, it merely registers with the event loop).
But it was originally buried under the other errors.

Reply:
More specifically for future reference: If shutdown is called before the event loop returns, there is a possibility the node's API gets called again **after** shutdown was called. The node should not be used after `shutdown()`.